src/app/modules/main/profile_section/contacts/module.nim

```diff
@@ -110,6 +110,18 @@ method acceptContactRequest*(self: Module, publicKey: string, contactRequestId:
 method dismissContactRequest*(self: Module, publicKey: string, contactRequestId: string) =
   self.controller.dismissContactRequest(publicKey, contactRequestId)
 
+method getLatestContactRequestForContactAsJson*(self: Module, publicKey: string): string =
+  let contactRequest = self.controller.getLatestContactRequestForContact(publicKey)
+  let jsonObj = %* {
+    "id": contactRequest.id,
+    "from": contactRequest.from,
+    "parsedText": contactRequest.parsedText,
+    "text": contactRequest.text,
+    "contactRequestState": contactRequest.contactRequestState.int,
+    "seen": contactRequest.seen
```

Comment:
Looks good but it could be even simplified to just `timestamp` when the CR was sent (not sure if `seen` matches that?) and the `challenge` text (which one of those `text` and `parsedText` should I take? :)

Reply:
fixed